mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergSerDe.java

```diff
@@ -56,10 +62,22 @@ public void initialize(@Nullable Configuration configuration, Properties serDePr
     } else if (serDeProperties.get(InputFormatConfig.TABLE_SCHEMA) != null) {
       tableSchema = SchemaParser.fromJson((String) serDeProperties.get(InputFormatConfig.TABLE_SCHEMA));
     } else {
-      try {
-        tableSchema = Catalogs.loadTable(configuration, serDeProperties).schema();
-      } catch (NoSuchTableException nte) {
-        throw new SerDeException("Please provide an existing table or a valid schema", nte);
+      // Read the configuration parameters
+      String columnNames = serDeProperties.getProperty(serdeConstants.LIST_COLUMNS);
+      String columnTypes = serDeProperties.getProperty(serdeConstants.LIST_COLUMN_TYPES);
```

Comment:
Are these properties set on table as properties, or are they generated dynamically during reads for column pruning?
If they are set as table properties, what happens if they go stale with respect to the iceberg table? e.g. column renames, type promotions, new top level fields added, etc.

Reply:
Hive does a lot of magic here:
- First creates a SerDe to get the columns - in this case these properties are empty
- Uses the objectInspectors from the SerDe to update column specification of the table.
- Then when it does the query planning it again calls this part, but the properties are already set based on the result of the previous call

With that sad this theoretically solves the schema evolution problem, but I am not convinced that we caught all of the nuances. So further down the road we plan to test this out as well and file the new PRs as required.

Reply:
So If I understand correctly, this saves us from having to load the table from the catalog multiple times during query compile/planning.

Reply:
Yes, and also this is the way how we get the information when the columns are provided in the create table query.
```
CREATE EXTERNAL TABLE (customer_id INT, first_name STRING) customers 
STORED BY 'org.apache.iceberg.mr.hive.HiveIcebergStorageHandler';
```
Will result in:
```
columnNames = "customer_id,first_name";
columnTypes = "int:string";
```
Don't ask me how Hive arrived to this solution, but it is how it is...